src/java.base/share/classes/javax/crypto/EncryptedPrivateKeyInfo.java

```diff
@@ -304,15 +315,167 @@ private PKCS8EncodedKeySpec getKeySpecImpl(Key decryptKey,
+     * provider list.
+     * <p>
+     * The PBE algorithm string format details can be found in the
+     * <a href="{@docRoot}/../specs/security/standard-names.html#cipher-algorithms">
+     * Cipher section</a> of the Java Security Standard Algorithm Names
+     * Specification.
+     *
+     * @param key the PrivateKey object to encrypt.
+     * @param password the password used for generating the PBE key.
+     * @param algorithm the PBE encryption algorithm.
+     * @param params the parameters used with the PBE encryption.
+     * @param provider the Provider that will perform the encryption.
+     * @return an EncryptedPrivateKeyInfo.
+     * @throws IllegalArgumentException when an argument causes an
+     * initialization error.
+     * @throws SecurityException on a cryptographic errors.
+     * @throws NullPointerException if the key, password, or algorithm are null.
+     * null.
+     *
+     * @since 24
+     */
+    @PreviewFeature(feature = PreviewFeature.Feature.PEM_API)
+    public static EncryptedPrivateKeyInfo encryptKey(PrivateKey key,
+        char[] password, String algorithm, AlgorithmParameterSpec params,
+        Provider provider) {
+
+        PBEKeySpec keySpec = new PBEKeySpec(password);
+        Objects.requireNonNull(key);
+        Objects.requireNonNull(algorithm);
+        Cipher cipher;
+        SecretKey skey;
+
+        try {
+            SecretKeyFactory factory;
+            if (provider == null) {
+                factory = SecretKeyFactory.getInstance(algorithm);
+                cipher = Cipher.getInstance(algorithm);
+            } else {
+                factory = SecretKeyFactory.getInstance(algorithm, provider);
+                cipher = Cipher.getInstance(algorithm, provider);
+            }
+            skey = factory.generateSecret(keySpec);
+        } catch (NoSuchAlgorithmException | NoSuchPaddingException |
+                 InvalidKeySpecException e) {
+            throw new IllegalArgumentException(e);
+        }
+
+        AlgorithmId algId;
+        byte[] encryptedData;
+        DerOutputStream out = new DerOutputStream();
+        try {
+            cipher.init(Cipher.ENCRYPT_MODE, skey, params);
+            encryptedData = cipher.doFinal(key.getEncoded());
+            algId = new AlgorithmId(Pem.getPBEID(algorithm),
+                cipher.getParameters());
+            algId.encode(out);
+            out.putOctetString(encryptedData);
+        } catch (InvalidAlgorithmParameterException e) {
+            throw new IllegalArgumentException(e);
+        } catch (IllegalBlockSizeException | BadPaddingException |
+                 InvalidKeyException e) {
+            throw new SecurityException(e);
+        }
+        return new EncryptedPrivateKeyInfo(
+            DerValue.wrap(DerValue.tag_Sequence, out).toByteArray(),
+            encryptedData, algId, cipher.getParameters());
+    }
+
+    /**
+     * Creates and encrypts an `EncryptedPrivateKeyInfo` from a given PrivateKey
```

Comment:
Oh, you should use `{@code}` here. Same in `@impNote` below. Also, there are some other class names that should be in enclosed in `{@code}`.

Reply:
ok